test/EVMHost.cpp

```diff
@@ -771,14 +771,141 @@ evmc::result EVMHost::precompileALTBN128G1Mul(evmc_message const& _message) noex
 
 evmc::result EVMHost::precompileALTBN128PairingProduct(evmc_message const& _message) noexcept
 {
-	// Input must be divisible by 192.
-	// TODO: validate points
-	if (_message.input_size % 192)
-		return resultWithFailure();
-
-	// This is a partial implementation - it always returns "success"
-	bytes static data = fromHex("0000000000000000000000000000000000000000000000000000000000000001");
-	return resultWithGas(_message, data);
+	// NOTE this is a partial implementation for some inputs.
+	static map<bytes, bytes> const inputOutput{
+		{
+			fromHex(
+				"17c139df0efee0f766bc0204762b774362e4ded88953a39ce849a8a7fa163fa9"
+				"01e0559bacb160664764a357af8a9fe70baa9258e0b959273ffc5718c6d4cc7c"
+				"198e9393920d483a7260bfb731fb5d25f1aa493335a9e71297e485b7aef312c2"
+				"1800deef121f1e76426a00665e5c4479674322d4f75edadd46debd5cd992f6ed"
+				"090689d0585ff075ec9e99ad690c3395bc4b313370b38ef355acdadcd122975b"
+				"12c85ea5db8c6deb4aab71808dcb408fe3d1e7690c43d37b4ce6cc0166fa7daa"
+				"0000000000000000000000000000000000000000000000000000000000000001"
+				"30644e72e131a029b85045b68181585d97816a916871ca8d3c208c16d87cfd45"
+				"0a09ccf561b55fd99d1c1208dee1162457b57ac5af3759d50671e510e428b2a1"
+				"2e539c423b302d13f4e5773c603948eaf5db5df8ae8a9a9113708390a06410d8"
+				"19b763513924a736e4eebd0d78c91c1bc1d657fee4214057d21414011cfcc763"
+				"2f8d9f9ab83727c77a2fec063cb7b6e5eb23044ccf535ad49d46d394fb6f6bf6"
+			),
+			fromHex("0000000000000000000000000000000000000000000000000000000000000001")
```

Comment:
Do we have any tests that actually use these outputs?

Reply:
Ah right. Somehow I thought it must have been failing without these but now I realized that we were just accepting any input. In that case if the CI passes, it must be fine.